Web3/Classes/Core/Providers/AtomicCounter.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  AtomicCounter.swift
+//  Web3
+//
+//  Created by Yehor Popovych on 3/16/19.
+//
+
+import Foundation
+
+public class AtomicCounter {
```

Comment:
Awesome. Thanks.

Reply:
I will update this class in the future when they will stabilize Atomic APIs (they want to use llvm based approach). Right now NSLock based (works, but not so efficient as native atomic)

Reply:
You just have to make sure it also works on Linux. I am not sure about the "NS" APIs...

Reply:
It works 100%. They implemented NSLock (via pthread methods)

Reply:
Yes. The CI also succeeded so it can at least compile it...But could you still add some Tests for this class?

Reply:
Will try to find time for tests. I need to split PR too :)